objecthash.py

```diff
@@ -194,10 +199,8 @@ def redactable_key(k):
     return redactable_rand() + k
 
 def redactable_rand():
-    r = ''
-    for x in range(32):
-        r += chr(random.SystemRandom().getrandbits(8))
-    return hexify(r)
+    r = random.SystemRandom().getrandbits(256)
+    return hex(r)
```

Comment:
Ugh, wait a second. I was trying to avoid messing around with types with this change, but this change doesn't ensure the returned str is of length 32.